std/algebra/emulated/sw_emulated/point.go

```diff
@@ -758,59 +858,64 @@ func (c *Curve[B, S]) jointScalarMulGLV(p1, p2 *AffinePoint[B], s1, s2 *emulated
 func (c *Curve[B, S]) jointScalarMulGLVUnsafe(Q, R *AffinePoint[B], s, t *emulated.Element[S]) *AffinePoint[B] {
 	var st S
 	frModulus := c.scalarApi.Modulus()
-	sd, err := c.scalarApi.NewHint(decomposeScalarG1, 5, s, c.eigenvalue, frModulus)
+	sd, err := c.scalarApi.NewHint(decomposeScalarG1, 7, s, c.eigenvalue, frModulus)
 	if err != nil {
 		// err is non-nil only for invalid number of inputs
 		panic(err)
 	}
-	s1, s2, s3, s4 := sd[0], sd[1], sd[3], sd[4]
+	s1, s2, s3, s4, s5, s6 := sd[0], sd[1], sd[3], sd[4], sd[5], sd[6]
 
-	td, err := c.scalarApi.NewHint(decomposeScalarG1, 5, t, c.eigenvalue, frModulus)
+	td, err := c.scalarApi.NewHint(decomposeScalarG1, 7, t, c.eigenvalue, frModulus)
 	if err != nil {
 		// err is non-nil only for invalid number of inputs
 		panic(err)
 	}
-	t1, t2, t3, t4 := td[0], td[1], td[3], td[4]
+	t1, t2, t3, t4, t5, t6 := td[0], td[1], td[3], td[4], td[5], td[6]
 
 	c.scalarApi.AssertIsEqual(
-		c.scalarApi.Add(s1, c.scalarApi.Mul(s2, c.eigenvalue)),
+		c.scalarApi.Add(s5, c.scalarApi.Mul(s6, c.eigenvalue)),
 		c.scalarApi.Add(s, c.scalarApi.Mul(frModulus, sd[2])),
 	)
 	c.scalarApi.AssertIsEqual(
-		c.scalarApi.Add(t1, c.scalarApi.Mul(t2, c.eigenvalue)),
+		c.scalarApi.Add(t5, c.scalarApi.Mul(t6, c.eigenvalue)),
 		c.scalarApi.Add(t, c.scalarApi.Mul(frModulus, td[2])),
 	)
-	// s1, s2 can be negative (bigints) in the hint, but are reduced mod r in
-	// the circuit. So we return in the hint both s1, s2 and s3=|s1|, s4=|s2|.
-	// In-circuit we compare s1 and s3, s2 and s4 and negate the point when a
-	// corresponding scalar is naegative. Respectively for t1, t2, t3, t4.
-	selector1 := c.scalarApi.IsZero(c.scalarApi.Sub(s3, s1))
-	selector2 := c.scalarApi.IsZero(c.scalarApi.Sub(s4, s2))
-	selector3 := c.scalarApi.IsZero(c.scalarApi.Sub(t3, t1))
-	selector4 := c.scalarApi.IsZero(c.scalarApi.Sub(t4, t2))
+
+	// s1, s2 can be negative (bigints) in the hint, but will be reduced
+	// in-circuit modulo the SNARK scalar field and not the emulated field. So
+	// we return in the hint both |s1|, |s2| and the flags s3=0/1, s4=0/1 to
+	// negate the point instead of the corresponding scalar. Since s3, s4 are
+	// either 0 or 1, we only need to check the first limb is zero or not.
+	// Same goes for t1, t2.
+	selector1 := c.api.IsZero(s3.Limbs[0])
+	selector2 := c.api.IsZero(s4.Limbs[0])
+	selector3 := c.api.IsZero(t3.Limbs[0])
+	selector4 := c.api.IsZero(t4.Limbs[0])
```

Comment:
Same comment that should assert that `sd1, sd3, sd5` etc. should correspond to each other.